pkg/scheduler/plugins/elasticquota/core/hook_plugin.go

```diff
@@ -0,0 +1,294 @@
+}
+
+func (w *MetricsWrapper) PostQuotaUpdate(oldQuotaInfo, newQuotaInfo *QuotaInfo, quota *v1alpha1.ElasticQuota, state *QuotaUpdateState) {
+	start := time.Now()
+	w.plugin.PostQuotaUpdate(oldQuotaInfo, newQuotaInfo, quota, state)
+	metrics.RecordElasticQuotaHookPluginLatency(w.GetKey(), "post_quota_update", time.Since(start))
+}
+
+func (w *MetricsWrapper) OnPodUpdated(quotaName string, oldPod, newPod *v1.Pod) {
+	start := time.Now()
+	w.plugin.OnPodUpdated(quotaName, oldPod, newPod)
+	metrics.RecordElasticQuotaHookPluginLatency(w.GetKey(), "on_pod_update", time.Since(start))
+}
+
+func (w *MetricsWrapper) UpdateQuotaStatus(oldQuota, newQuota *v1alpha1.ElasticQuota) *v1alpha1.ElasticQuota {
+	start := time.Now()
+	result := w.plugin.UpdateQuotaStatus(oldQuota, newQuota)
+	metrics.RecordElasticQuotaHookPluginLatency(w.GetKey(), "update_quota_status", time.Since(start))
+	return result
+}
+
+func (w *MetricsWrapper) CheckPod(quotaName string, pod *v1.Pod) error {
+	start := time.Now()
+	err := w.plugin.CheckPod(quotaName, pod)
+	metrics.RecordElasticQuotaHookPluginLatency(w.GetKey(), "check_pod", time.Since(start))
+
+	return err
+}
+
+func (w *MetricsWrapper) GetPlugin() QuotaHookPlugin {
+	return w.plugin
+}
+
+// The following methods in GroupQuotaManager handle interactions with hook plugins
+
+func (gqm *GroupQuotaManager) InitHookPlugins(pluginArgs *config.ElasticQuotaArgs) error {
+	hookPlugins, err := initHookPlugins(gqm.GetQuotaInfoReader(), pluginArgs)
+	if err != nil {
+		return fmt.Errorf("failed to init hook plugins for tree %s, err=%v", gqm.GetTreeID(), err)
+	}
+	gqm.SetHookPlugins(hookPlugins)
+	return nil
+}
+
+func (gqm *GroupQuotaManager) GetHookPlugins() []QuotaHookPlugin {
+	return gqm.hookPlugins
+}
+
+func (gqm *GroupQuotaManager) SetHookPlugins(hookPlugins []QuotaHookPlugin) {
+	// Wrap plugins with metrics
+	gqm.hookPlugins = WrapWithMetrics(hookPlugins)
+}
+
+func (gqm *GroupQuotaManager) GetQuotaInfoReader() *QuotaInfoReader {
+	return &QuotaInfoReader{
+		GetQuotaInfo:             gqm.GetQuotaInfoByName,
+		GetChildQuotaInfos:       gqm.GetChildGroupQuotaInfos,
+		GetQuotaInfoNoLock:       gqm.getQuotaInfoByNameNoLock,
+		GetChildQuotaInfosNoLock: gqm.getChildGroupQuotaInfosNoLock,
+	}
+}
+
+func (gqm *GroupQuotaManager) GetChildGroupQuotaInfos(quotaName string) map[string]*QuotaInfo {
+	gqm.hierarchyUpdateLock.RLock()
+	defer gqm.hierarchyUpdateLock.RUnlock()
+
+	return gqm.getChildGroupQuotaInfosNoLock(quotaName)
+}
+
+func (gqm *GroupQuotaManager) getChildGroupQuotaInfosNoLock(quotaName string) map[string]*QuotaInfo {
+	quotaTopoNode := gqm.quotaTopoNodeMap[quotaName]
+	if quotaTopoNode == nil || len(quotaTopoNode.getChildGroupQuotaInfos()) == 0 {
+		return nil
+	}
+	childQuotaInfos := make(map[string]*QuotaInfo)
+	for childName, childTopoNode := range quotaTopoNode.getChildGroupQuotaInfos() {
+		childQuotaInfos[childName] = childTopoNode.quotaInfo
```

Comment:
这里还是从quotaInfoMap取quota info比较靠谱

Reply:
Done. Thanks for the review. 
The latest commit also added Key field in HookPluginConf since it's more reasonable to return Key instead of FactoryKey when calling HookPlugin#GetKey().